src/twister2/log_file/log_file_abstract.py

```diff
@@ -0,0 +1,39 @@
+from __future__ import annotations
+
+import abc
+import logging
+import os
+import sys
+from pathlib import Path
+
+logger = logging.getLogger(__name__)
+
+
+class LogFileAbstract(abc.ABC):
+    """Base class for logging files."""
+
+    def __init__(self, build_dir: Path | str, name: str) -> None:
+        self.default_encoding = sys.getdefaultencoding()
+        self.filename = self._get_log_filename(build_dir=build_dir, name=name)
+
+    @classmethod
+    def _get_log_filename(cls, build_dir: Path | str, name: str) -> str:
```

Comment:
Is there any particular reason to use there `@classmethod`? Can it be used just as a method? Like this:
```python
def _get_log_filename(self, build_dir: Path | str, name: str) -> str:
    ...
    filename = self._normalize_filename_path(filename=filename)
    ...
```

Reply:
Done